### PR TITLE
Fix for #44 Reqnroll in RootNamespace breaks generated Files

### PR DESCRIPTION
**Pull request by clrudolphi**
This fix changes all references to Reqnroll types in the generated code as global:: references.

Created a utility method in CodeDomHelper to take a Type and return a string as "global::" + type full name.

Replaced all occurrences of typeof(<some Reqnroll type>) with an invocation of the utility method. This switches the use of the CodeDom Type Reference constructors from using a Type to using a String (should have no noticeable side-effects).
Replaced a few occurrences where the generators were already using the string name of the type.

Tests are provided for a simple scenario and one containing a DataTable (as suggested).
Don't yet have tests for DocStrings, Scenario Outlines, Rules or Backgrounds. Welcome suggestions on which of these to do and how to proceed (using a resource file?).

Not sure what, if anything, needs to be done to support VB.

## Types of changes

- [X ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Performance improvement
- [ ] Refactoring (so no functional change)
- [ ] Other (docs, build config, etc)

## Checklist:

- [X ] I've added tests for my code. (most of the time mandatory)
- [X ] I have added an entry to the changelog. (mandatory)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by gasparnagy** [comment]
@clrudolphi looks good.

Notes for the system tests: 

* This problem is related to "generation" (as it generates invalid code), so I would move the test to the `GenerationTestBase` class. That way it is even ensured that it is checked with all 3 test execution framework. The project name setup line (`_projectsDriver.CreateProject(...)`) you can simply into the test (before the call to `AddScenario`). 
* These tests are very "expensive" so you should follow different guidelines when deciding on the tests than for unit tests. In this case, the two tests can be simply merged: a single scenario that has a data table step is actually checking both cases anyway. (But if needed we could add multiple steps for the single scenario, or even have multiple `AddScenario` call within the same test.)

It seems from the CI that a few of the unit tests in Reqnroll.GeneratorTests needs to be adjusted. Please check that and click on the "Ready for review" if you think it is ready. I will do a final review after that.

File: Reqnroll.Generator/CodeDom/CodeDomHelper.cs [hunk 1]
```diff
@@ -287,6 +287,17 @@ private CodeExpression GetAwaitedMethodThisTargetObject(CodeExpression thisExpre
...
         }
+
+        public string GetGlobalizedTypeName(Type type)
+        {
+            if (TargetLanguage == CodeDomProviderLanguage.CSharp)
+            {
+                return "global::" + type.FullName!;
+            }
+
+            //TODO: VB.NET
```
Comment (gasparnagy) [inline Reqnroll.Generator/CodeDom/CodeDomHelper.cs:298]: I would not leave this TODO in the code. It is unlikely that we will deal too much with VB. We can keep a comment, saying that we do not support global namespaces for VB, but not a TODO.

**Review by gasparnagy — CHANGES_REQUESTED** [review]
Thank you! I've just made a small request.

File: Reqnroll.Generator/CodeDom/CodeDomHelper.cs (hunk 1 above)
Comment (clrudolphi) [reply Reqnroll.Generator/CodeDom/CodeDomHelper.cs]: Agreed. 
Modification made.

**Review by gasparnagy — APPROVED** [review]

**Comment by gasparnagy** [comment]
@clrudolphi thank you!

Do you have any feedback on how system tests worked? Any challenges you have faced?

**Comment by clrudolphi** [comment]
The only minor hiccup was that AddScenario() creates a (default) feature, so that you cannot call AddScenario more than one time per test.

It is a little challenging to get control over feature structure and feature and file naming but not insurmountable. Use of the 'load feature from resource' might be the better option for those testing situations.

Once I had the prerequisite framework versions installed, all the tests ran smoothly and pretty quickly.

**Comment by clrudolphi** [comment]
@gasparnagy Should we also cover situations in which the user has placed the name of the testing framework (such as XUnit) in their namespace? See for example this bug report on SpecFlow [here](https://github.com/SpecFlowOSS/SpecFlow/issues/2742)

**Pull request merged**